### PR TITLE
dragon-drop: remove duplicate package

### PR DESCRIPTION
**Pull request by anpryl**
###### Description of changes

Removing package because it is a duplicate package of "dragon".
Added replacement warning for it.
Discussion: https://github.com/NixOS/nixpkgs/issues/162874

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -2982,9 +2982,7 @@ with pkgs;
 
   dpic = callPackage ../tools/graphics/dpic { };
 
-  dragon-drop = callPackage ../tools/X11/dragon-drop {
-    gtk = gtk3;
-  };
+  dragon-drop = throw "'dragon-drop' has been renamed to/replaced by 'xdragon'"; # Converted to throw 2022-03-27;
```
Comment (bobby285271) [inline pkgs/top-level/all-packages.nix:2985]: Maybe put a `dragon-drop = xdragon; # Added 2022-03-27` line in `pkgs/top-level/aliases.nix` instead? We have [scripts](https://github.com/NixOS/nixpkgs/blob/master/maintainers/scripts/remove-old-aliases.py) to convert alias to throw.

Comment (anpryl) [reply pkgs/top-level/all-packages.nix:2985]: Fixed

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (ajs124) [reply pkgs/top-level/all-packages.nix:2985]: On one hand, I agree with @bobby285271. On the other hand, the package about the be dropped provides a binary called `dragon`, whereas the one that will remain provides a binary called `xdragon`, for reasons outlined in https://github.com/NixOS/nixpkgs/pull/95925 or rather https://github.com/mwh/dragon/issues/17

So throw might be the better option, after all.

Comment (thiagokokada) [reply pkgs/top-level/all-packages.nix:2985]: I also think throwing maybe a better option in this case.

**Pull request merged**